runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -357,6 +377,65 @@ private void translateKinesisInput(
             .addSource(source, FlinkLyftKinesisConsumer.class.getSimpleName()));
   }
 
+  private void translateS3AndKinesisInputs(
+      String id,
+      RunnerApi.Pipeline pipeline,
+      FlinkStreamingPortablePipelineTranslator.StreamingTranslationContext context) {
+    RunnerApi.PTransform pTransform = pipeline.getComponents().getTransformsOrThrow(id);
+
+    ObjectMapper mapper = new ObjectMapper();
+    String programConfigPath;
+    JSONObject jsonContent;
+
+    try {
+      JsonNode params = mapper.readTree(pTransform.getSpec().getPayload().toByteArray());
+      Preconditions.checkNotNull(
+          programConfigPath = params.path(
+              "program_config_path").textValue(),
+          "'program_config_path' needs to be set");
+
+
+      File programConfigFile = new File(programConfigPath);
+      if (!programConfigFile.exists()) {
+        throw new IllegalArgumentException("File not found at " + programConfigPath);
+      }
+
+      InputStream inputStream = new FileInputStream(programConfigPath);
+      String fileContent = IOUtils.toString(inputStream, "UTF-8");
+      jsonContent = new JSONObject(fileContent);
+
+      Preconditions.checkNotNull(jsonContent, "Program config file seems to be empty");
+
+    } catch (IOException e) {
+      throw new RuntimeException("Could not parse config");
+    }
+
+    ProgramConfiguration programConfig = ProgramConfiguration.parse(jsonContent);
+    StreamExecutionEnvironment streamExecutionEnvironment = LyftStreamExecutionEnvironment.get();
```

Comment:
This should not appear here. The execution environment is managed by the Flink runner.